storage/acls.rb

```diff
@@ -0,0 +1,281 @@
+  puts "\tOWNER"  if bucket.acl.owners.include?  email
+  puts "\tWRITER" if bucket.acl.writers.include? email
+  puts "\tREADER" if bucket.acl.readers.include? email
+  # [END print_bucket_acl_for_user]
+end
+
+def add_bucket_owner project_id:, bucket_name:, email:
+  # [START add_bucket_owner]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # email       = "Google CLoud Storage ACL Entity email"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.acl.add_owner email
+
+  puts "Added OWNER permission for #{email} to #{bucket_name}"
+  # [END add_bucket_owner]
+end
+
+def remove_bucket_acl project_id:, bucket_name:, email:
+  # [START remove_bucket_acl]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # email       = "Google CLoud Storage ACL Entity email"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.acl.delete email
+
+  puts "Removed ACL permissions for #{email} from #{bucket_name}"
+  # [END remove_bucket_acl]
+end
+
+def add_bucket_default_owner project_id:, bucket_name:, email:
+  # [START add_bucket_default_owner]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # email       = "Google CLoud Storage ACL Entity email"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.default_acl.add_owner email
+
+  puts "Added default OWNER permission for #{email} to #{bucket_name}"
+  # [END add_bucket_default_owner]
+end
+
+def remove_bucket_default_acl project_id:, bucket_name:, email:
+  # [START remove_bucket_default_acl]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # email       = "Google CLoud Storage ACL Entity email"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+
+  bucket.default_acl.delete email
+
+  puts "Removed default ACL permissions for #{email} from #{bucket_name}"
+  # [END remove_bucket_default_acl]
+end
+
+def print_file_acl project_id:, bucket_name:, file_name:
+  # [START print_file_acl]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Storage bucket"
+  # email       = "Google CLoud Storage ACL Entity email"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+  file    = bucket.file file_name
+
+  puts "ACL for #{file_name} in #{bucket_name}:"
+
+  file.acl.owners.each do |owner|
```

Comment:
Nope, only OWNER and READER is available per the gem[1] and documentation[2].
> WRITER -> N/A. You cannot apply this permission to objects.


1. https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-storage/lib/google/cloud/storage/file/acl.rb#L77
2. https://cloud.google.com/storage/docs/access-control/lists#permissions